src/rt-app_parse_config.c

```diff
@@ -902,13 +961,18 @@ parse_task_data(char *name, struct json_object *obj, int index,
 	data->def_cpu_data.cpuset = NULL;
 	data->def_cpu_data.cpuset_str = NULL;
 
-	data->curr_sched_data = NULL;
-
 	/* cpuset */
 	parse_cpuset_data(obj, &data->cpu_data);
 	/* Scheduling policy */
 	data->sched_data = parse_sched_data(obj, opts->policy);
 
+	/*
+	 * Thread's current sched_data/taskgroup are used to detect
+	 * policy/taskgroup misconfiguration.
+	 */
+	data->curr_sched_data = data->sched_data;
```

Comment:
HAve you check that this change in the initial state of curr_sched_data doesn't impact the behavior of sched policy ?
we have this condition ``
	if (data->curr_sched_data == sched_data)
		return;
`` in set_thread_priority() which means that we will never apply the 1st policy or the task policy

Reply:
Like you've already noted, I reset data->curr_sched_data to NULL at the end of the function.